ext/oci8/oci8_failover.c

```diff
@@ -101,30 +99,33 @@ int php_oci_disable_taf_callback(php_oci_connection *connection)
 int php_oci_register_taf_callback(php_oci_connection *connection, char *callback)
 {
 	sword errstatus;
-	char *oldCallback = NULL;
+	int registered = 0;
 
 	/* temporary failover callback structure */
 	OCIFocbkStruct failover;
 
 	if (!callback) {
 		/* Disable callback */
-		if (!connection->taf_callback || !strcmp(PHP_OCI_TAF_DISABLE_CALLBACK, connection->taf_callback)) {
+		if (Z_ISUNDEF(connection->taf_callback)) {
 			return 0; // Nothing to disable
 		}
 
-		oldCallback = connection->taf_callback;
-		callback = PHP_OCI_TAF_DISABLE_CALLBACK;
-	} else if (connection->taf_callback) {
-		/* Overwriting old callback */
-		oldCallback = connection->taf_callback;
-	}
+		registered = 1;
+		zval_ptr_dtor(&connection->taf_callback);
+		ZVAL_UNDEF(&connection->taf_callback);
+	} else {
+		if (!Z_ISUNDEF(connection->taf_callback)) {
```

Comment:
If it's set to null, it should still set registered to 1, so instead I would change the block at line 117 to:
```
if (!Z_ISUNDEF(connection->taf_callback)) {
    registered = 1;
    if (!Z_ISNULL(connection->taf_callback)) {
        zval_ptr_dtor(&connection->taf_callback);
        ZVAL_NULL(&connection->taf_callback);
    }
}
```